docs/qprofiler2/index.md

```diff
@@ -0,0 +1,328 @@
+--threads-producer <Integer>  Opt, number of threads to read the indexed BAM file, BAM mode only. This  
+                                option will be ignored without "threads-consumer" option.  Def=1.       
+--vcf-format-field            Opt, group VCF records according to user specified format fields. eg.     
+                                "GD", "CCM","FT=PASS". Default to ignore VCF format field.              
+--version                     Show version number.     
+~~~~
+
+
+### multi thread
+When running multi-threaded, we suggest more consumer than producer threads with a recommended ratio of 6:1. However, it is up to your machine system. for example, only one thread to read the input file but 12 threads are specified to process reads.
+
+Please specify the BAM index file if multiple producer threads are going to be used, eg. 
+
+~~~~{.text}
+java -jar qprofiler2.jar --threads-consumer 12 -threads-producer 2 --bam-index  $somedir/${bam}.bai --input  $somedir/$bam --output $somedir/${bam}.qp2.xml --log $somedir/${bam}.qp2.log
+~~~~
+
+## Output
+
+### Xml Validation
+
+qprofiler2 provide a schema file which help you to validate the xml output. This xsd file can download from https://purl.org/adamajava/xsd/qprofiler2/v2/qprofiler2.xsd
+
+~~~~{.text}
+xmllint --noout --schema ~/PATH/qprofier2.xsd qprofiler2.xml
+or
+java -jar xsd11-validator.jar -sf ~/PATH/qprofier2.xsd  -if qprofiler2.xml
+~~~~
+
+### FASTQ Mode output
+FASTQ mode outputs three sections information: read name analysis, sequence analysis and base quality analysis. An example as below:
+
+~~~~{.text}
+<qProfiler finish_time=2021-09-10 16:59:32 run_by_os=Linux run_by_user=christiX start_time=2021-09-10 16:51:25 version=78-43982c9>
+  <fastqReport execution_finished=2021-09-10 16:59:32 execution_started=2021-09-10 16:51:25 file=/mnt/Illumina_gdc_realn_1.fastq.gz records_parsed=86,324,947>
+   <ReadNameAnalysis>
+     <INSTRUMENTS>...</INSTRUMENTS>
+     <RUN_IDS>...</RUN_IDS>
+     <FLOW_CELL_IDS/>
+     <FLOW_CELL_LANES>...</FLOW_CELL_LANES>
+     <TILE_NUMBERS>...</TILE_NUMBERS>
+     <PAIR_INFO>...</PAIR_INFO>
+     <FILTER_INFO>...</FILTER_INFO>
+     <INDEXES/>
+     <QUAL_HEADERS>...</QUAL_HEADERS>
+   </ReadNameAnalysis>
+   <SEQ>
+     <BaseByCycle>...</BaseByCycle>
+     <LengthTally>...</LengthTally>
+     <BadBasesInReads>...</BadBasesInReads>
+     <mers6>...</mers6>
+     <mers1>...</mers1>
+     <mers2>...</mers2>
+     <mers3>...</mers3>
+   </SEQ>
+   <QUAL>
+     <QualityByCycle>...</QualityByCycle>
+     <LengthTally>...</LengthTally>
+     <BadQualsInReads>
+     <ValueTally>...</ValueTally>
+     </BadQualsInReads>
+   </QUAL>
+</fastqReport>
+</qProfiler>
+
+~~~~
+
+### BAM Mode output
+BAM record contains both fastq record information and algnment information. qProfiler2 outputs based on BAM record element order, but also output BAM header and summary information. An example as below:
```

Comment:
"qProfiler2 outputs based on BAM record element order, but also output BAM header and summary information" - what is this sentence trying to say?